org.eclipse.jdt.apt.pluggable.core/src/org/eclipse/jdt/internal/apt/pluggable/core/filer/IdeNonSourceOutputStream.java

```diff
@@ -48,40 +45,20 @@ public IdeNonSourceOutputStream(IdeProcessingEnvImpl env, IFile file, Collection
 	public void close() throws IOException {
 		super.close();
 
-		InputStream contents = new ByteArrayInputStream(toByteArray());
+		byte[] newContent = toByteArray();
+		boolean contentsChanged = true;
 		try {
-
-			boolean contentsChanged = true;
-			if (!_file.exists()) {
-				saveToDisk(contents, true);
-			}
-			else {
-				InputStream in = null;
-				InputStream oldData = null;
-				try {
-					// Only write the contents if the data is different
-					in = new ByteArrayInputStream(toByteArray());
-					oldData = new BufferedInputStream(_file.getContents());
-					if (FileSystemUtil.compareStreams(in, oldData)) {
-						contentsChanged = false;
-					}
-				}
-				catch (CoreException ce) {
-					// Ignore -- couldn't read the old data, so assume it's different
-					contentsChanged = true;
-				}
-				finally {
-					closeInputStream(in);
-					closeInputStream(oldData);
-				}
-				if (contentsChanged) {
-					contents.reset();
-					saveToDisk(contents, false);
-				}
+			// Only write the contents if the data is different
```

Comment:
The fact this is a copy of the previous code is an existing issue.

Reply:
org.eclipse.jdt.internal.apt.pluggable.core.filer.IdeNonSourceOutputStream,
org.eclipse.jdt.internal.apt.pluggable.core.filer.IdeClassOutputStream,
org.eclipse.jdt.apt.core.internal.env.BinaryFileOutputStream,

are now refactored to use common FileSystemUtil.saveToDisk(IFile, byte[])